### ISSUE TITLE
Virtual license scaling

### ISSUE THREAD

**Issue by wstuebing** [MicrosoftDocs/OfficeDocs-SkypeForBusiness#2700]
Hello,

I am wondering what is going to be done for organizations that had hundreds of Auto Attendants and call Queues that do not meet this sliding scale. 

For example you had 100 Auto Attendants under Skype for business that were migrated over. When the only restriction was on number assigned via the portal for service. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d6891495-01c6-117b-c28b-28232782ba74
* Version Independent ID: 1c23b35a-e9ba-ffdd-9baa-a3f83f4ed9ca
* Content: [Virtual user licenses ](https://docs.microsoft.com/en-US/microsoftteams/teams-add-on-licensing/virtual-user)
* Content Source: [Teams/teams-add-on-licensing/virtual-user.md](https://github.com/MicrosoftDocs/OfficeDocs-SkypeForBusiness/blob/live/Teams/teams-add-on-licensing/virtual-user.md)
* Service: **msteams**
* GitHub Login: @jambirk
* Microsoft Alias: **jambirk**

**Comment by jambirk** [comment]
The Virtual user license (or Phone System License) is only needed for a resource account linked to an auto attendant node that has a phone number assigned. In large deployments with dozens of nested Auto Attendants, the number of dial-in numbers is still relatively limited, You could easily have a deployment that had a hundred auto attendants and call queues total, but only three or four ingress points/phone numbers, thus using a handful of virtual user licenses.

**Comment by wstuebing** [comment]
Thanks @jambirk for the comment the issues is that I have 97 public facing auto attendants that do not use nested auto attendants.  As well as 71 call queue that are mix of numbers and some without. Most have a number as they were setup in Skype online before the teams admin change. 

Based on this scale I only get access to 94 Virtual licenses for having 690 users with phone system. 

This model works great for business that have 3 or 4 locations with many users in each site but if you have branch offices then this scale does not work. The fix is to grandfather organizations in that had migrated accounts from SFB online or adjust the scale.  We did not have any issues until this move to Teams admin center. As you were able request service numbers based on total phone users. Or port in your service numbers and this had no effect on your counts.

**Comment by tonysmit** [comment]
Jumping in here James to help too.
@wstuebing -  You do have a point here but it would be interesting to find out how many enterprises have this set up. That said, you would be short 3 virtual licenses plus the other ones for the CQs that have a direct phone number. 

@jambirk - Let's give the PG this feedback. Connect with the PMs.
CC: @waseemhashem @nomorephones

**Comment by wstuebing** [comment]
@tonysmit - Normally we just ride this out until everything is normalized but this change is right at are True up True down so it is a bit of an issue.

**Comment by jambirk** [comment]
@wstuebing also keep in mind that phone system user licenses would still work on the resource accounts. The Virtual User license might be preferable because it's free, but if you need more than is provided for in the scaling as described, then the phone system user license is a completely viable solution.

**Comment by wstuebing** [comment]
@jambirk - Can one assign a phone system licence to an account without and E1? Same with consumption billing for toll free they still require a Calling plan - Phone or Vphone - Consumption?  Id test this but due to the true up we are over assigned as Microsoft removed are last large reservation with the E1 - Phone - Calling Plan that was needed.

**Comment by tonysmit** [comment]
@wstuebing - I cc'd the PMs on this, so I'm sure that they have taken this feedback, but I will ping them to make sure they are notified. I believe both are on vacation, so there might be a delay on an answer from them. 

So, just so I'm clear here. The UX does a license check to check for the existence of a PS license (either flavor). For this, it only checks that one exists. I've confirmed that with the dev. That said, for RAs to work, **A** (either can be assigned) phone system license (that might not be the same license) must be assigned to the RA. The requirement **isn't** that an M/O365 license is also assigned. 

I've sent a message to the PMs to figure out what the remediation would be for this scenario. I will report back or I will talk to @jambirk, we are on the same team and he can reply. 

Thank you for pointing this out. @jambirk owns the topic and will update it as soon as we hear back.  Thanks, Tony

**Comment by tonysmit** [comment]
@wstuebing - I forgot to ask on your reply to @jambirk. Yes, the RA doesn't have to have an O365 license assigned (either a Small Business Essentials, Office Preminum or an E-based). However, if you have an E5 licenses (with PS), you can assign this full license (althought this would be expensive) because it includes a PS license. 

Now, as far as the calling plan and communication credits. Communication credits are required, but a calling plan is not for toll free service numbers.

**Comment by wstuebing** [comment]
@tonysmit and @jambirk  Thanks for all the help on this.  We did get are RA accounts setup with Phone System- Virtual user and removed the Domestic calling plan from most at it is not needed.  In doing so I was able to try and setup an RA account with Phone System - Domestic calling plan - Communication credits and this will not apply as it required a E1 as well so i have left it on Phone system Virtual user.
